### PR TITLE
feat: completion for interactive mode

### PR DESCRIPTION
**Pull request by stuartwdouglas**
fixes: #2708

### CODE REVIEW

File: frontend/cli/cmd_interactive.go [hunk 1]
```diff
@@ -9,19 +9,28 @@ import (
...
 	"github.com/TBD54566975/ftl/internal/projectconfig"
 )
 
+var _ readline.AutoCompleter = &FTLCompletion{}
+
 type interactiveCmd struct {
 }
 
-func (i *interactiveCmd) Run(ctx context.Context, k *kong.Kong, projectConfig projectconfig.Config) error {
+type FTLCompletion struct {
```
Comment (alecthomas) [inline frontend/cli/cmd_interactive.go:24]: I know I said to put all the types at the top, but I'm going to slightly contradict myself slightly 😂

It's preferable to have type followed immediately by methods. So in this case I'd put FTLCompletion + methods at the top, then interactiveCmd + methods below that, so it still works to read down the file and have seen everything.

Not a big deal though, but just for your reference.

File: frontend/cli/cmd_interactive.go [hunk 2]
```diff
@@ -62,3 +74,72 @@ func (i *interactiveCmd) Run(ctx context.Context, k *kong.Kong, projectConfig pr
...
+	if parser == nil {
+		return nil, 0
+	}
+	all := []string{}
+	completed := []string{}
+	last := ""
+	lastCompleted := ""
+	lastSpace := false
+	// We don't care about anything past pos
+	// this completer can't handle completing in the middle of things
```
Comment (alecthomas) [inline frontend/cli/cmd_interactive.go:89]: Yeah I've noticed that before, it sucks a bit :( - still, better than no completion

File: frontend/cli/cmd_interactive.go [hunk 3]
```diff
@@ -62,3 +74,72 @@ func (i *interactiveCmd) Run(ctx context.Context, k *kong.Kong, projectConfig pr
 func errorf(format string, args ...any) {
 	fmt.Printf("\033[31m%s\033[0m\n", fmt.Sprintf(format, args...))
 }
+
+func (f *FTLCompletion) Do(line []rune, pos int) ([][]rune, int) {
```
Comment (alecthomas) [inline frontend/cli/cmd_interactive.go:78]: Awesome!

**Review by alecthomas — APPROVED** [review]
So cool!

File: frontend/cli/cmd_interactive.go (hunk 1 above)
Comment (stuartwdouglas) [reply frontend/cli/cmd_interactive.go:24]: That is where I was going to put it, but I remembered your comment from yesterday :-) I will move it.

**Pull request merged**